### PR TITLE
Add 'provenance.submission_date' and '.update_date' fields (#45)

### PR DESCRIPTION
**Pull request by dsotirho-ucsc**

### CODE REVIEW

**Comment by dsotirho-ucsc** [comment]
Note that according to the https://schema.humancellatlas.org/system/1.1.0/provenance schema, `submission_date` is a required field and `update_date` is optional.

File: test/test.py [hunk 1]
```diff
@@ -356,11 +357,30 @@ def test_file_core_bundle(self):
...
+            '2019-09-20T08:29:52.247Z': 1,
+            '2019-09-20T08:29:52.277Z': 1,
+            '2019-09-20T08:29:52.285Z': 1,
+            '2019-09-20T08:29:52.292Z': 1,
+            '2019-09-20T08:29:52.299Z': 1,
+        })
+        update_date_counts = Counter(entity.update_date
+                                     for entity in bundle.files.values())
+        self.assertEqual(update_date_counts, {
+            None: 1,
```
Comment (jessebrennan) [inline test/test.py:375]: Assuming that this missing one is because of the deletion up above, right?

**Review by jessebrennan — APPROVED** [review]

File: test/test.py (hunk 1 above)
Comment (dsotirho-ucsc) [reply test/test.py:375]: Yes, I removed the `update_date` field from one of the metadata files of the canned bundle to make sure the case of the optional field being not present was covered.

File: test/test.py [hunk 2]
```diff
@@ -356,11 +357,30 @@ def test_file_core_bundle(self):
                                    library_construction_methods={"10x 3' v3 sequencing"},
                                    content_description={'DNA sequence'})
         submitter_id_counts = Counter(file.submitter_id for file in bundle.files.values())
-        expected_counts = {
+        self.assertEqual(submitter_id_counts, {
```
Comment (hannes-ucsc) [inline test/test.py]: Expected should be first argument. And why do you inline one argument but extract the other into a variable? Both are relatively long expressions.

File: test/test.py [hunk 3]
```diff
@@ -356,11 +357,30 @@ def test_file_core_bundle(self):
...
             None: 3,  # Sequence files without a submitter_id field
             '67a720af-4482-4619-81d7-3693b2d3cc4c': 4  # Supplementary files with submitter_id
-        }
-        self.assertEqual(expected_counts, submitter_id_counts)
+        })
+        submission_date_counts = Counter(entity.submission_date
+                                         for entity in bundle.files.values())
+        self.assertEqual(submission_date_counts, {
+            '2019-09-20T08:29:52.232Z': 1,
```
Comment (hannes-ucsc) [inline test/test.py]: Might want to comprehend over this since only the millis are different.

File: test/test.py [hunk 4]
```diff
@@ -356,11 +357,30 @@ def test_file_core_bundle(self):
...
+            '2019-09-20T08:29:52.277Z': 1,
+            '2019-09-20T08:29:52.285Z': 1,
+            '2019-09-20T08:29:52.292Z': 1,
+            '2019-09-20T08:29:52.299Z': 1,
+        })
+        update_date_counts = Counter(entity.update_date
+                                     for entity in bundle.files.values())
+        self.assertEqual(update_date_counts, {
+            None: 1,
+            '2019-09-20T08:38:58.165Z': 2,
```
Comment (hannes-ucsc) [inline test/test.py:376]: This one might not be worth a comprehension.

File: src/humancellatlas/data/metadata/api.py [hunk 5]
```diff
@@ -120,6 +120,8 @@ def __init__(self,
         if False and self.metadata_manifest_entry is not None:
             assert self.document_id == self.metadata_manifest_entry.uuid
         self.submitter_id = provenance.get('submitter_id')
+        self.submission_date = lookup(provenance, 'submission_date', 'submissionDate')
+        self.update_date = lookup(provenance, 'update_date', 'updateDate', default=None)
```
Comment (hannes-ucsc) [inline src/humancellatlas/data/metadata/api.py:126]: These attributes should be declared at the top.

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**